dashboard/apps/dashboard/spreadsheets.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+import re
+
+from openpyxl.styles import Style, Font
+from openpyxl.workbook import Workbook
+
+from dashboard.libs.date_tools import parse_date
+
+
+class Products:
+
+    date_style = Style(number_format='DD/MM/YYYY')
+    bold_font = Font(bold=True)
+    bold_style = Style(font=bold_font)
+    currency_style = Style(number_format='£#,##0.00')
+
+    def __init__(self, products):
+        self.workbook = Workbook()
+        self.fill_main_sheet(self.workbook.active, products)
+        if len(products) == 1:
+            # only create monthly spend sheet when there is one product
+            self.fill_monthly_spend_sheet(self.workbook.create_sheet(), products[0])
+
+    @staticmethod
+    def format(value):
+        if isinstance(value, date):
+            return value.strftime('%d/%m/%Y')
+        return value
+
+    @staticmethod
+    def kwarg_vals(kwargs):
+        vals = []
+        for k, v in kwargs.items():
+            if k == 'year':
+                v = '%s-%s' % (str(v)[2:], str(v + 1)[2:])
+            vals.append(v)
+        return vals
```

Comment:
this method is moved from the old code in the `view.py` module with little refactoring. most of these comments are valid and reflect that this code needs a bit rethinking. i'll just go ahead and rewrite it.